atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/pue/ResilientCommitTimestampPutUnlessExistsTable.java

```diff
@@ -100,10 +101,9 @@ public void putUnlessExistsMultiple(Map<Long, Long> keyValues) throws KeyAlready
                                 + "was found in the KVS",
                         SafeArg.of("kvsValue", kvsValue),
                         SafeArg.of("stagingValue", currentValue));
-                continue;
+            } finally {
+                resultBuilder.put(startTs, commitTs);
```

Comment:
This seems a bit odd. If an exception is thrown, won't this appear to a caller that a value was committed, but in reality it has not?

Reply:
If a CheckAndSetException is thrown here, it means the value was committed by another thread, and hence we must return it.

If other exceptions are thrown, then we will propagate the exception out of this method, and thus would not return the result anyway.

This is quite subtle, though, so I'll add a clarifying comment.